content/sensu-go/5.17/guides/secrets-management.md

```diff
@@ -0,0 +1,289 @@
+---
+title: "Use secrets management in Sensu"
+linkTitle: "Use Secrets Management"
+description: "Sensu's secrets management allows you to avoid exposing secrets like usernames and passwords in your Sensu configuration. In this guide, you'll learn how to use Sensu's built-in secrets provider or HashiCorp Vault and refer to external secrets in your Sensu configuration."
+weight: 175
+version: "5.17"
+product: "Sensu Go"
+platformContent: False
+menu: 
+  sensu-go-5.17:
+    parent: guides
+---
+
+- [Use Env for secrets management](#use-env-for-secrets-management)
+  - [Create your backend environment variable](#create-your-backend-environment-variable)
+  - [Create your secret](#create-your-secret)
+- [Use HashiCorp Vault for secrets management](#use-hashicorp-vault-for-secrets-management)
+  - [Retrieve your Vault root token](#retrieve-your-vault-root-token)
+  - [Create your Vault secrets provider](#create-your-vault-secrets-provider)
+  - [Create your secret](#create-your-secret)
+- [Add a handler](#add-a-handler)
+  - [Register the PagerDuty handler asset](#register-the-pagerduty-handler-asset)
+  - [Add your secret to the handler spec](#add-your-secret-to-the-handler-spec)
+- [Next steps](#next-steps)
+
+**COMMERCIAL FEATURE**: Access the Env and VaultProvider secrets provider datatypes in the packaged Sensu Go distribution.
+For more information, see [Get started with commercial features][20].
+
+Sensu's secrets management allows you to avoid exposing secrets like usernames, passwords, and access keys in your Sensu configuration.
+In this guide, you'll learn how to use Sensu's built-in secrets provider, `Env`, or [HashiCorp Vault][1] as your external [secrets provider][2] and authenticate without exposing your secrets
+
+To follow this guide, you’ll need to [install the Sensu backend][5], have at least one [Sensu agent][11] running, and [install and configure sensuctl][7].
+
+Secrets are configured via [secrets resources][8].
+A secret resource definition refers to the secrets provider (`Env` or `VaultProvider`) and an ID (the named secret to fetch from the secrets provider).
+
+This guide explains how to set up your [PagerDuty Events API Integration Key][27] as a secret and a PagerDuty handler that requires the secret.
+Your Sensu backend can then execute the handler with any check.
+The Sensu backend will transmit requests over its secure transport (TLS-encrypted websockets) to your Sensu agent to execute your handler.
+
+**Is this true for Vault? Even though we're using the no-TLS development server?**
```

Comment:
Yes, all secrets regardless of provider are only transmitted to agents if the agent is using TLS.